### ISSUE TITLE
[Kotlin 1.7.20-Beta] IllegalStateException: Couldn't find debug in...

### ISSUE THREAD

**Issue by jimgoog** [square/wire#2315]
After upgrading from `Kotlin 1.7.10` to `Kotlin 1.7.20-Beta`, we started getting an exception of the following form:
```
Caused by: java.lang.IllegalStateException: Couldn't find debug in {androidTest=null, androidTestDebug=null, androidTestRelease=null, debug=null, main=null, release=null, test=null, testDebug=null, testFixtures=null, testFixturesDebug=null, testFixturesRelease=null, testRelease=null}
        at com.squareup.wire.gradle.kotlin.SourceRootsKt.sourceRoots(SourceRoots.kt:149)
        at com.squareup.wire.gradle.kotlin.SourceRootsKt.sourceRoots(SourceRoots.kt:77)
        at com.squareup.wire.gradle.WirePlugin$sources$2.invoke(WirePlugin.kt:46)
        at com.squareup.wire.gradle.WirePlugin$sources$2.invoke(WirePlugin.kt:46)
        at kotlin.SynchronizedLazyImpl.getValue(LazyJVM.kt:74)
        at com.squareup.wire.gradle.WirePlugin.getSources(WirePlugin.kt:46)
        at com.squareup.wire.gradle.WirePlugin.setupWireTasks(WirePlugin.kt:132)
        at com.squareup.wire.gradle.WirePlugin.access$setupWireTasks(WirePlugin.kt:38)
        at com.squareup.wire.gradle.WirePlugin$apply$androidPluginHandler$1.invoke$lambda-0(WirePlugin.kt:66)
        at org.gradle.configuration.internal.DefaultUserCodeApplicationContext$CurrentApplication$1.execute(DefaultUserCodeApplicationContext.java:123)
        at org.gradle.configuration.internal.DefaultListenerBuildOperationDecorator$BuildOperationEmittingAction$1.run(DefaultListenerBuildOperationDecorator.java:171)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute(DefaultBuildOperationRunner.java:29)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute(DefaultBuildOperationRunner.java:26)
```

Closest related error I could find is:
https://github.com/square/wire/issues/1864

Is there a build of this plugin which is compatible with `Kotlin 1.7.20-Beta`?

**Comment by oldergod** [comment]
Thanks for reporting. Probably fixed with #2310.
I'll try to release something soon.

**Comment by ibevilinc** [comment]
4.4.2 with kotlin 1.7.20 now breaks with the following exception
```
Caused by: java.lang.NullPointerException
        at com.squareup.wire.gradle.kotlin.SourceRootsKt.sourceRoots(SourceRoots.kt:149)
        at com.squareup.wire.gradle.kotlin.SourceRootsKt.sourceRoots(SourceRoots.kt:78)
        at com.squareup.wire.gradle.WirePlugin$sources$2.invoke(WirePlugin.kt:46)
        at com.squareup.wire.gradle.WirePlugin$sources$2.invoke(WirePlugin.kt:46)
        at kotlin.SynchronizedLazyImpl.getValue(LazyJVM.kt:74)
        at com.squareup.wire.gradle.WirePlugin.getSources(WirePlugin.kt:46)
        at com.squareup.wire.gradle.WirePlugin.setupWireTasks(WirePlugin.kt:132)
        at com.squareup.wire.gradle.WirePlugin.access$setupWireTasks(WirePlugin.kt:38)
        at com.squareup.wire.gradle.WirePlugin$apply$androidPluginHandler$1.invoke$lambda-0(WirePlugin.kt:66)
```

**Comment by oldergod** [comment]
I had trouble building the project last week so I wasn't able to make progress. I hopefully will have a fix this week.

**Comment by damianw** [comment]
Looked at this briefly - seems that the issue is [here](https://github.com/square/wire/blob/39a37ef06936a6fa2897c2f34494e9aa6b353202/wire-library/wire-gradle-plugin/src/main/kotlin/com/squareup/wire/gradle/kotlin/SourceSets.kt#L32-L40). `KotlinSourceSet` is no longer a direct superinterface for the class of the convention object. Is it doing it this way (finding another instance of the `Class` instead of using the given `Class`) because the `KotlinSourceSet` implemented on the object is from a different classloader?

**Comment by oldergod** [comment]
Fixed in #2340 which is part of the 4.4.3 release.
